libs/ts-rest/nest/src/lib/api.decorator.ts

```diff
@@ -6,24 +6,26 @@ import {
   Delete,
   ExecutionContext,
   Get,
-  HttpException,
   Injectable,
   NestInterceptor,
   Patch,
   Post,
   Put,
+  SetMetadata,
   UseInterceptors,
 } from '@nestjs/common';
 import {
   AppRoute,
   AppRouteMutation,
   checkZodSchema,
-  getPathParamsFromUrl,
   PathParamsWithCustomValidators,
   Without,
   ZodInferOrType,
 } from '@ts-rest/core';
 import { map, Observable } from 'rxjs';
+import type { Request, Response } from 'express-serve-static-core';
```

Comment:
is this correct in the resulting package.json (not pinned to a super specific version or anything?)

Reply:
@oliverbutler ideally it shouldn't be added to the package.json at all as it is only being used for typing during compile time, but I just checked it seems like it's being added to the peer dependencies. Let me see if there's a solution for that. For some reason `qs` is being added to core's package.json too despite being used only in a spec file.

Reply:
The qs problem seems to be an issue with nx https://github.com/nrwl/nx/issues/8096
I recently moved to turborepo because of nx's weird quirks

Reply:
Yeah, that's a tad weird. NX should be smart enough to purge it from the resulting json...

Also, I just noticed you removed the `getPathParamsFromUrl` helper, honestly, I can't remember _why_ this existed 🤔 It's probably because I wasn't sure how to access the params as you did!

Maybe we can delete the `libs/ts-rest/core/src/lib/server-utils.ts` in their entirely 🎉

Reply:
Yeah, let me dig into that nx thing.

As for `getPathParamsFromUrl` it was being used instead of `req.params` in nest, but it looks like it's still being used by next. I guess because the [...ts-rest] param needs to be expanded from a full URL into individual params.

Reply:
Ok, it should be fine now. Looks like nx isn't able to handle `qs` being imported using a wildcard `import * as qs...` so I've used the named imports instead. For `express-serve-static-core` I've just included it into dev deps so it's not added into the peer deps

Reply:
Ok, awesome 🔥 I'll investigate the next thing. I'll see if I can move that code out of the core lib to decrease bundle size.